### ISSUE TITLE
ruby installation not working

### ISSUE THREAD

**Issue by johnwahba** [grpc/homebrew-grpc#49]
@tbetbetbe for visibility

```
✗ curl -fsSL https://goo.gl/getgrpc | bash -s ruby
Warning: Already tapped!
zlib is already installed
openssl is already installed
Warning: Already tapped!
google-protobuf is already installed
grpc is already installed
installing grpc ruby
Fetching: grpc-0.10.0.gem (100%)
Building native extensions with: '--with-grpc-dir=/usr/local'
This could take a while...
ERROR:  Error installing grpc:
    ERROR: Failed to build gem native extension.

    /Users/$user/.rbenv/versions/2.1.5/bin/ruby -r ./siteconf20150715-86487-wtwbpd.rb extconf.rb --with-grpc-dir=/usr/local
checking for dlopen() in -ldl... yes
checking for inflate() in -lz... yes
checking for gpr_now() in -lgpr... yes
checking for grpc_channel_destroy() in -lgrpc... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling rb_byte_buffer.c
compiling rb_call.c
rb_call.c:134:66: error: missing field 'reserved' initializer [-Werror,-Wmissing-field-initializers]
    {GRPC_RB_GC_NOT_MARKED, GRPC_RB_GC_DONT_FREE, md_ary_datasize},
                                                                 ^
rb_call.c:142:78: error: missing field 'reserved' initializer [-Werror,-Wmissing-field-initializers]
    {GRPC_RB_GC_NOT_MARKED, grpc_rb_call_destroy, GRPC_RB_MEMSIZE_UNAVAILABLE},
                                                                             ^
rb_call.c:274:50: error: unused parameter 'key' [-Werror,-Wunused-parameter]
static int grpc_rb_md_ary_capacity_hash_cb(VALUE key, VALUE val,
                                                 ^
rb_call.c:349:64: error: unused parameter 'val' [-Werror,-Wunused-parameter]
static int grpc_rb_call_check_op_keys_hash_cb(VALUE key, VALUE val,
                                                               ^
4 errors generated.
make: *** [rb_call.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/$user/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/grpc-0.10.0 for inspection.
Results logged to /Users/$user/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/extensions/x86_64-darwin-14/2.1.0-static/grpc-0.10.0/gem_make.out

```

**Comment by tbetbetbe** [comment]
@murgatroid99 PTAL, this is probably related to something missing on the release branch?

**Comment by murgatroid99** [comment]
OK, I can reproduce this, but only on Mac. I suspect I know why: when doing pkg-config work with ruby, we also enabled compiler warnings. It worked just fine, but I guess we never tested it on a Mac.

**Comment by murgatroid99** [comment]
And on further examination, the warnings listed above are, or at least should be, enabled on Linux as well. I have no idea why those were not already caught.

**Comment by johnwahba** [comment]
@murgatroid99, It may have to do with the src/ruby/tmp/*/Makefile warnflags being respected on one platform and not on the other or it could be a difference in compilers.

**Comment by murgatroid99** [comment]
OK, I think I figured it out. This appears to be caused by a difference in how GCC and Clang (XCode) interpret repeated warning flags. And I think I can fix it.

**Comment by tbetbetbe** [comment]
This has been resolved by the beta release (0.11.0)
